monai/inferers/merger.py

```diff
@@ -0,0 +1,186 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from collections.abc import Sequence
+
+import torch
+
+from monai.utils import ensure_tuple_size
+
+__all__ = ["Merger", "AvgMerger"]
+
+
+class Merger(ABC):
+    """
+    A base class for merging patches.
+    Extend this class to support operations for `PatchInference`, e.g. `AvgMerger`.
+
+    Args:
+        device: the device where Merger tensors should reside.
+    """
+
+    def __init__(self, device: torch.device | str | None = None, dtype: torch.dtype = torch.float32) -> None:
+        self.device = device
+        self.dtype = dtype
+        self.is_initialized = False
+        self.is_finalized = False
+
+    @abstractmethod
+    def initialize(self, output_shape: Sequence[int] | None = None) -> None:
+        """
+        Initialize the merger.
+
+        Args:
+            output_shape: the shape of output tensor (B,C,H,W,[D])
+
+        Raises:
+            NotImplementedError: When the subclass does not override this method.
+
+        """
+        raise NotImplementedError(f"Subclass {self.__class__.__name__} must implement this method.")
+
+    @abstractmethod
+    def aggregate(self, values: torch.Tensor, location: Sequence[int]) -> None:
+        """
+        Aggregate values for merging.
+
+        Args:
+            values: a tensor of shape BCHW[D], representing the values of inference output
+            location: a tuple/list giving the top left location of the patch in the original image.
+
+        Raises:
+            NotImplementedError: When the subclass does not override this method.
+
+        """
+        raise NotImplementedError(f"Subclass {self.__class__.__name__} must implement this method.")
+
+    @abstractmethod
+    def finalize(self) -> torch.Tensor:
```

Comment:
This should state what the expected return value is meant to be. It's clear in `AvgMerger` what it does but it should say here in the general case.

Reply:
I update it to be Any for the base class and added more description what is expected to be returned in general.